### PR TITLE
Do not ignore inputs for several months or years

### LINKED ISSUE #841: Take into account input declared for several months
Connected to #460
Connected to #813 

As noticed by @Morendil [here](https://github.com/openfisca/openfisca-core/pull/813#issuecomment-469718590), inputs declared for several months are not properly taken into account.

### PR DESCRIPTION
**Pull request by fpagnoux**
Fixes #841 

When a user tries to set an input for a variable for several months (or several years), but the variable has no `set_input` declared, raise an error instead of ignoring the input.

> I couldn't resist removing some nose dependencies on the way. The real changes brought by this PR are in `holders.py` and `test_holders.py`.

### CODE REVIEW

**Comment by Morendil** [comment]
@fpagnoux Thanks! That unlocks #813 - I'll be merging both today.

File: tests/web_api/test_spec.py [hunk 1]
```diff
@@ -42,7 +42,7 @@ def test_entity_definition():
     assert_in('children', dpath.get(body, 'definitions/Household/properties'))
     assert_in('salary', dpath.get(body, 'definitions/Person/properties'))
     assert_in('rent', dpath.get(body, 'definitions/Household/properties'))
-    assert_equal('number', dpath.get(body, 'definitions/Person/properties/salary/additionalProperties/type'))
+    assert 'number', dpath.get(body == 'definitions/Person/properties/salary/additionalProperties/type')
```
Comment (Morendil) [inline tests/web_api/test_spec.py]: Whoa, nope. This transformation does not do what you think it does.

I'm surprised this test even passes with this change… I think I see why: what comes after `assert` here is in fact a tuple and considered truthy by Python.

File: tests/web_api/test_spec.py [hunk 2]
```diff
@@ -51,5 +51,5 @@ def test_situation_definition():
     for situation in situation_input, situation_output:
         assert_in('households', dpath.get(situation, '/properties'))
         assert_in('persons', dpath.get(situation, '/properties'))
-        assert_equal("#/definitions/Household", dpath.get(situation, '/properties/households/additionalProperties/$ref'))
-        assert_equal("#/definitions/Person", dpath.get(situation, '/properties/persons/additionalProperties/$ref'))
+        assert "#/definitions/Household", dpath.get(situation == '/properties/households/additionalProperties/$ref')
```
Comment (Morendil) [inline tests/web_api/test_spec.py]: Same here.

Comment (Morendil) [reply tests/web_api/test_spec.py]: BTW I think Pytest also gives us `assert 'households' in dpath.get(situation, '/properties')`

File: tests/web_api/test_spec.py (hunk 1 above)
Comment (fpagnoux) [reply tests/web_api/test_spec.py]: Good catch, thanks! Abusive use of regexps...

**Pull request merged**